src/vm/gdbjit.cpp

```diff
@@ -1291,13 +1290,46 @@ bool NotifyGdb::BuildDebugAbbrev(MemBuf& buf)
     return true;
 }
 
+int NotifyGdb::GetArgsAndLocalsLen(NewArrayHolder<ArgsDebugInfo>& argsDebug,
+                                   unsigned int argsDebugSize,
+                                   NewArrayHolder<LocalsDebugInfo>& localsDebug,
+                                   unsigned int localsDebugSize)
+{
+    int loc_size = 0;
+    char tmp_buf[16];
+    for (int i = 0; i < argsDebugSize; i++)
+    {
+        loc_size += Leb128Encode(static_cast<int32_t>(argsDebug[i].m_native_offset), tmp_buf, sizeof(tmp_buf));
+        loc_size += 2;
+    }
+    for (int i = 0; i < localsDebugSize; i++)
+    {
+        loc_size += Leb128Encode(static_cast<int32_t>(localsDebug[i].m_native_offset), tmp_buf, sizeof(tmp_buf));
+        loc_size += 2;
+    }
+    return loc_size;
+}
+
+int NotifyGdb::GetFrameLocation(int native_offset, char*& bufVarLoc)
```

Comment:
Why is the bufVarLoc passed by reference when it is not changed by the function?